src/main/java/kotlinx/reflect/lite/impl/KPropertyImpl.kt

```diff
@@ -20,7 +20,7 @@ internal abstract class KPropertyImpl<out T>(
     override val isSuspend: Boolean
         get() = false
 
-    override fun call(vararg args: Any?): T = getter.call(args)
+    override fun call(vararg args: Any?): T = getter.call(*args)
```

Comment:
This one is really a bummer: without this change, only `accessPrivateProperties` has been falling, meaning that there is not a single test on `call` on getter. 

Could you please add some from `unusedTestData` and ensure that they fail without this change, right in this branch?

Reply:
Also, this test (`accessPrivateProperties`) started to fail for an irrelevant reason: because I've added the same caching as in `kotlin-reflect`.

For some reason, after that, `getBar === bar` is true, even though `kotlinReflectGetBar === kotlinReflectBar` is false (moreover, they are different types). It does not seem expected, please investigate it, post results here and add a test (or, ideally, find one in unusedTestData)

Reply:
Filed https://youtrack.jetbrains.com/issue/KTIJ-22644/Inspection-for-missing-spread-operator

Reply:
Before we added caching: `bar` and `getBar` properties were not equal and setting accessibility of the `javaField` corresponding to the `bar` did not affect the `javaField` of `getBar` property. So, `getBar.call(A())` [was failing with the expected error](https://github.com/Kotlin/kotlinx.reflect.lite/blob/23297f144bc2eda7d2ca6ae0aeadaa17c23f29e4/src/test/java/kotlinx/reflect/lite/testData/call/accessPrivateProperties.kt#L28) because of the inaccessibility of the member. The check for accessibility is performed before calling, so we didn't catch the bug with the wrong argument type.

After caching: `bar` and `getBar` point to the same `KMutableProperty1<A, String>` property :` (A::class.java).kotlin` does not create the new `KClass` instance now and we get the same member both ways. So, setting `isAccessible` of the javaField of this property allowed to make a call to the property and the test started to fail with the wrong argument error.

Reply:
And `bar` and `getBar` are equal because we reference the same property in the members of the class now

Reply:
Nice, great job with figuring it out!

I have a question -- do we **semantically** expect `bar` and `getBar` to be the same or is it just a limitation of current approach?

Reply:
Yes, I guess, they should be equal